utilities/backupable/backupable_db.cc

```diff
@@ -181,6 +197,8 @@ class BackupEngineImpl : public BackupEngine {
     const std::string filename;
     const uint64_t size;
     const std::string checksum_hex;
+    const std::string custom_checksum_hex;
+    const std::string checksum_func_name;
```

Comment:
Do you think we need to use custom_checksum_func_name?

Reply:
`checksum_func_name` always gives priority to custom checksum function names. That is, when a custom checksum function is used, `checksum_func_name` is always the name of the custom checksum function. Otherwise, it is the name of the backup default. Since there is only one backup default and if it is used, `checksum_func` will be null, we do not need another variable for the name here.

A related question might be: do we need `checksum_func_name`?
I think could remove it (although I prefer keeping it) because it mainly serves as an in-memory sanity-check (or for the purpose of debugging) since we do not write `checksum_func_name` to the backup meta file. If the checking of `checksum_func_name` failed, something seriously wrong should have happened. 

This also reminds me another question I had in mind. Currently, when `LoadFromFile()` is called, backup engine only reads the backup meta files as before and it does not read the backed up db manifests. (Because this will make the initialization of this version's backup engine slower, which might not be a good idea.) Backup engine will read the corresponding db manifest for a specific backup id when it is needed for verification/restoration.

If we choose to read the backed up db manifests when opening the backup engine, then the usage of `checksum_func_name` will be more than in-memory sanity-check. It may help detect corruption of shared table files that have been backed up.

Reply:
Agree, better to keep the `checksum_func_name` for now. Not a high overhead.